client-app/src/App.js

```diff
@@ -1,22 +1,27 @@
-import './App.css';
-import {createBrowserRouter, RouterProvider} from "react-router-dom";
-import RegisterPage from "./views/Register"
-import {Container} from "@mui/material";
+import "./App.css";
+import { createBrowserRouter, RouterProvider } from "react-router-dom";
+import RegisterPage from "./pages/Register.jsx";
+import { Container } from "@mui/material";
+import Login from "./pages/Login.jsx";
 
 function App() {
-    const router = createBrowserRouter([
-        {
-            path: "/",
-            element: <RegisterPage/>
-        }
-    ])
-    return (
-        <div className="App">
-            <Container style={{paddingTop: 5}}>
-                <RouterProvider router={router}/>
-            </Container>
-        </div>
-    );
+  const router = createBrowserRouter([
```

Comment:
I recently had a problem on my own project with `createBrowserRouter` and `RouterProvider`. The problem was that you could only use the navigation hooks in side the `RouterProvider`. This means we can't have components that persists between pages, and have those components use the navigation hooks. This is a problem if we want to create a navbar or a sidebar.

Instead I would ditch the `createBrowserRouter` for just a simple array of routes. (Basically have the parameter given to `createBrowserRouter` as its own variable). For the `RouterProvider` I would map the routes array we created into `<Routes>` components. For last would warp the entire thing in `BrowserRouter`. That would allow us to have a navigation element that persists on all pages.

The end code would look something like this:
```            
<BrowserRouter>
    <MyNavbar/>
    <Routes>
        {MyRoutes.map((route,i) =>
            <Route path={route.path} element={route.element} key={i}/>
        )}
    </Routes>
</BrowserRouter>
```